cms/src/main/java/uk/nhs/digital/externalstorage/workflow/action/ExternalFilePublishAction.java

```diff
@@ -0,0 +1,34 @@
+package uk.nhs.digital.externalstorage.workflow.action;
+
+import org.apache.commons.scxml2.SCXMLExpressionException;
+import org.apache.commons.scxml2.model.ModelException;
+import org.onehippo.cms7.services.HippoServiceRegistry;
+import org.onehippo.repository.documentworkflow.action.AbstractDocumentTaskAction;
+
+import uk.nhs.digital.externalstorage.s3.S3Connector;
+import uk.nhs.digital.externalstorage.workflow.task.ExternalFilePublishTask;
+
+public class ExternalFilePublishAction extends AbstractDocumentTaskAction<ExternalFilePublishTask> {
+
+    private static final long serialVersionUID = 1L;
+
+    public String getUnpublishedVariant() {
+        return getParameter("unpublishedVariantExpr");
+    }
+
+    @SuppressWarnings("unused")
+    public void setUnpublishedVariant(String unpublishedVariant) {
+        setParameter("unpublishedVariantExpr", unpublishedVariant);
+    }
+
+    @Override
+    protected ExternalFilePublishTask createWorkflowTask() {
+        return new ExternalFilePublishTask(HippoServiceRegistry.getService(S3Connector.class));
```

Comment:
I'd put the `HippoServiceRegistry.getService(S3Connector.class)` in initTask not in the constructor